src/wutil/dir_iter.rs

```diff
@@ -465,6 +481,9 @@ fn test_dir_iter() {
 
         // Links should never have a fast type if we are resolving them, since we cannot resolve a
         // symlink from readdir.
+        #[cfg(not(cygwin))]
+        // Symbolic links on Windows are complicated. Their behavior depends
```

Comment:
I think there's no need to repeat the same comment to 3 places, the commit message attached to the `#[cfg(not(cygwin))]` line is enough for me.

 > // Symbolic links on Windows are complicated. Their behavior depends
 > // on the CYGWIN or MSYS env variable

if it were just for environment variables, we could probably work
around that, i.e. we could override them in tests
(unless that's frowned upon).
But you mention that file systems limitations are another reason,
so probably there's enough strong reasons for disabling the test.